In [1]:
import pandas as pd 

In [2]:
df = pd.read_csv("data.csv", lineterminator='\n')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18135612 entries, 0 to 18135611
Data columns (total 2 columns):
 #   Column    Dtype 
---  ------    ----- 
 0   Key       int64 
 1   Sentence  object
dtypes: int64(1), object(1)
memory usage: 276.7+ MB


In [4]:
df.head()

Key                                           Sentence
0    0  $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHT...
1    1                                             Yahoo!
2    2                                    enjoy your oils
3    3                                               NAKK
4    4                              Congratulations girl!

In [5]:
df.tail()

Key                        Sentence
18135607  19332855    sng nk team meeting lepas ni
18135608  19332856                   thanks seller
18135609  19332857  Beautiful and healthy products
18135610  19332858           Seller is helpful too
18135611  19332859             Will purchase again

In [6]:
def pre_clean(df):
    # drop null
    df = df[df['Sentence'].notna()]
    # drop duplicates 
    df = df.drop_duplicates(subset='Sentence', keep="first")
    return df

In [8]:
data = pre_clean(df)
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9855006 entries, 0 to 18135609
Data columns (total 2 columns):
 #   Column    Dtype 
---  ------    ----- 
 0   Key       int64 
 1   Sentence  object
dtypes: int64(1), object(1)
memory usage: 225.6+ MB


# BEFORE VS AFTER PREPROCESSING

- before preprocessing: 5,701,763 
- after preprocessing: 9,855,006 

## 9.8 million 

In [9]:
data.head()

Key                                           Sentence
0    0  $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHT...
1    1                                             Yahoo!
2    2                                    enjoy your oils
3    3                                               NAKK
4    4                              Congratulations girl!

In [11]:
data = data[['Sentence']]

In [12]:
data.head()

Sentence
0  $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHTAG$ $HASHT...
1                                             Yahoo!
2                                    enjoy your oils
3                                               NAKK
4                              Congratulations girl!

In [15]:
from sklearn.utils import shuffle 

shuffled = shuffle(data)
shuffled.head()

Sentence
7861976                                        Lawa belaka
3179334                      kain microfibres cepat kering
5755734  Terbaikkkkkkkkkkkkk bagusssssssssssssss cunnnn...
4443364  Semoga motor saya tidak kesejukkan dimalam har...
6275930          Very fast delivery although from overseas

In [24]:
# Split the dataset for training and testing
cutoff = int(.20 * len(shuffled))
training_sentences = shuffled[:cutoff]
test_sentences = shuffled[cutoff:]

In [25]:
training_sentences.head()

Sentence
7861976                                        Lawa belaka
3179334                      kain microfibres cepat kering
5755734  Terbaikkkkkkkkkkkkk bagusssssssssssssss cunnnn...
4443364  Semoga motor saya tidak kesejukkan dimalam har...
6275930          Very fast delivery although from overseas

In [30]:
training_sentences.to_csv("data_train.csv", index=False)

In [27]:
test_sentences.head()

Sentence
10855980              it stay on your lips
6947526                Brang smp dgan cpat
16152310       keseluruhan servis amat bik
2183949   saya dh pkai 3 hari tanpa charge
9630432             Ada tali sekali diberi

In [31]:
test_sentences.to_csv("data_test.csv", index=False)

In [29]:
print("training size : ", len(training_sentences))
print("testing size : ", len(test_sentences))

training size :  1971001
testing size :  7884005


# split manually tagging file into 1k 

In [1]:
import pandas as pd

In [2]:
manual_tag_data = pd.read_csv("data_train.csv")

In [3]:
manual_tag_data.head()

Sentence
0                                        Lawa belaka
1                      kain microfibres cepat kering
2  Terbaikkkkkkkkkkkkk bagusssssssssssssss cunnnn...
3  Semoga motor saya tidak kesejukkan dimalam har...
4          Very fast delivery although from overseas

In [4]:
import math
import numpy as np

In [5]:
sen_list = manual_tag_data.Sentence.tolist()
sen_list_len = len(sen_list)

In [6]:
mm = math.ceil(sen_list_len/1000)

In [7]:
print(mm)

1972


In [8]:
sen_splitting = np.array_split(sen_list,mm)

In [9]:
count = 0
for sen_l in sen_splitting:
    df = pd.DataFrame(sen_l)
    df.columns = ['Sentence']
    df.to_csv('manual_tag_data/manual_' + str(count) +'.csv', index=False)
    count+=1

print(count)

1972


In [ ]:
import pandas as pd 
import math
import numpy as np

def mw_splitting():
    """
    This function aims to split the large unvalidate token into smaller chunk files.
    Each file will store a maximum of 1000 tokens only in csv file format.
    """
    df = pd.read_csv('token/en.csv')
    en_list = df.text.tolist()
    en_list_len = len(en_list)
    mm = math.ceil(en_list_len/1000) # split into <=1000 token per array 
    en_splitting = np.array_split(en_list,mm)
    
    # loop through the list of list 
    # should output 388 file in this case
    count = 0
    for en_l in en_splitting:
        df = pd.DataFrame(en_l)
        df.columns = ['text']
        df.to_csv('token_en_split/en_' + str(count) +'.csv', index=False)
        count+=1
 